[DOCS-7373] Add full-text search (#22333)
DataDog/documentation: content/en/logs/explorer/search_syntax.md

| `OR`         | **Union**: either term is contained in the selected events                                             | authentication OR password   |
| `-`          | **Exclusion**: the following term is NOT in the event (apply to each individual raw text search)                                                  | authentication AND -password |

> ### Full-text search across all log attributes

Author: ```suggestion
## Full-text search 
```

> <div class="alert alert-warning">The full-text search feature is only available in Log Management and works in monitor, dashboard, and notebook queries. The full-text search syntax cannot be used for defining index filters, archive filters, log pipeline filters, or in Live Tail. </div>

Reviewer: ```suggestion
<div class="alert alert-warning">The full-text search feature is only available in Log Management and works in monitor, dashboard, and notebook queries. The full-text search syntax cannot be used to define index filters, archive filters, log pipeline filters, or in Live Tail. </div>
```

Use the syntax `*:search_term` to perform a full-text search across all log attributes, including the log message.

> #### Single term example

Author: ```suggestion
### Single term example
```

| Search syntax | Search type | Description                                           |
| ------------- | ----------- | ----------------------------------------------------- |
| `*:hello` | Full-text   | Searches all log attributes for the term `hello`.     |
| `hello`       | Free text   | Searches only the log message for the term `hello`.   |

> #### Search term with wildcard example

Author: ```suggestion
### Search term with wildcard example
```

| Search syntax | Search type | Description                                                                                  |
| ------------- | ----------- | -------------------------------------------------------------------------------------------- |
| `*:hello` | Full-text   | Searches all log attributes for the exact string `hello`.                                    |
| `*:hello*`| Full-text   | Searches all log attributes for strings that starts with `hello`. For example, `hello_world`.|

> #### Multiple terms with exact match example

Author: ```suggestion
### Multiple terms with exact match example
```

| Search syntax       | Search type | Description                                            |
| ------------------- | ----------- |------------------------------------------------------- |

> | `*:"hello world"` | Full-text   |Searches all log attributes for the term `hello world`. |

Reviewer: ```suggestion
| `*:"hello world"` | Full-text   | Searches all log attributes for the term `hello world`. |
```

> | `hello world`       | Free text   |Searches only the log message for the term `hello`.     |

Reviewer: ```suggestion
| `hello world`       | Free text   | Searches only the log message for the term `hello`.     |
```

> #### Multiple terms without exact match example

Author: ```suggestion
### Multiple terms without exact match example
```

The full-text search syntax `*:hello world` is equivalent to `*:hello *:world`. It searches all log attributes for the terms `hello` and `world`.

> #### Multiple terms with a white space example

Author: ```suggestion
### Multiple terms with a white space example
```

service:*mongo
```

Wildcard searches can also be used to search in the plain text of a log that is not part of a log attribute. For example, this query returns all logs with content (message) that contain the string `NETWORK`:

```
*NETWORK*
```

However, this search term does not return logs that contain the string `NETWORK` if it is in a log attribute and not part of the log message.

### Full-text search across all log attributes

<div class="alert alert-warning">The full-text search feature is only available in Log Management and works in monitor, dashboard, and notebook queries. The full-text search syntax cannot be used for defining index filters, archive filters, log pipeline filters, or in Live Tail. </div>

<div class="alert alert-warning">You can use full-text search for all logs ingested after MONTH DATE YEAR.</div>

Use the syntax `*:search_term` to perform a free text search across all log attributes, including the log message.

#### Single term example

| Search syntax | Search type | Description                                           |
| ------------- | ----------- | ----------------------------------------------------- |

> | `*:("hello")` | Full-text   | Searches all log attributes for the term `hello`.     |

Author: ```suggestion
| `*:hello` | Full-text   | Searches all log attributes for the term `hello`.     |
```

| `hello`       | Free text   | Searches only the log message for the term `hello`.   |

#### Search term with wildcard example

| Search syntax | Search type | Description                                                                                  |
| ------------- | ----------- | -------------------------------------------------------------------------------------------- |

> | `*:("hello")` | Full-text   | Searches all log attributes for the exact string `hello`.                                    |
> | `*:("hello"*)`| Full-text   | Searches all log attributes for strings that starts with `hello`. For example, `hello_world`.|

Author: ```suggestion
| `*:hello` | Full-text   | Searches all log attributes for the exact string `hello`.                                    |
| `*:hello*`| Full-text   | Searches all log attributes for strings that starts with `hello`. For example, `hello_world`.|
```

#### Multiple terms with exact match example

| Search syntax       | Search type | Description                                            |
| ------------------- | ----------- |------------------------------------------------------- |

> | `*:("hello world")` | Full-text   |Searches all log attributes for the term `hello world`. |
> | `hello world`       | Free text   |Searches only the log message for the term `hello`.     |

Author: ```suggestion
| `*:hello world` | Full-text   |Searches all log attributes for the term `hello world`. |
| `hello world`       | Free text   |Searches only the log message for the term `hello`.     |
```

service:*mongo
```

Wildcard searches can also be used to search in the plain text of a log that is not part of a log attribute. For example, this query returns all the logs that contain the string `NETWORK`:

```
*NETWORK*
```

However, this search term does not return logs that contain the string `NETWORK` if it is in a log attribute and not part of the log message.

### Full-text search across all log attributes

> <div class="alert alert-warning">The full-text search feature is in beta. It is only available in Log Management and works for monitors, dashboards, and notebooks. The full-text search syntax cannot be used for defining index filters, archive filters, log pipeline filters, or in Live Tail. </div>

Author: ```suggestion
<div class="alert alert-warning">The full-text search feature is in beta. It is only available in Log Management and works in monitor, dashboard, and notebook queries. The full-text search syntax cannot be used for defining index filters, archive filters, log pipeline filters, or in Live Tail. </div>
```

> <div class="alert alert-warning">The full-text search feature is in beta. It is only available in Log Management and works in monitor, dashboard, and notebook queries. The full-text search syntax cannot be used for defining index filters, archive filters, log pipeline filters, or in Live Tail. </div>

Author: > works in monitor, dashboard, and notebook queries.

@siddhingra can you confirm this is correct?

service:*mongo
```

> Wildcard searches can also be used to search in the plain text of a log that is not part of a log attribute. For example, this query returns all the logs that contain the string `NETWORK`:

Reviewer 2: ```suggestion
Wildcard searches can also be used to search in the plain text of a log that is not part of a log attribute. For example, this query returns all the logs with content (message) that contain the string `NETWORK`:
```

```
*NETWORK*
```

However, this search term does not return logs that contain the string `NETWORK` if it is in a log attribute and not part of the log message.

### Full-text search across all log attributes

> <div class="alert alert-warning">The full-text search feature is in beta. It is only available in Log Management and works in monitor, dashboard, and notebook queries. The full-text search syntax cannot be used for defining index filters, archive filters, log pipeline filters, or in Live Tail. </div>

Reviewer 2: I no longer believe this is beta. It was shipped to everyone as of March 25 2024 AFAIK / AFAICT: https://dd.slack.com/archives/C057S9CHMSB/p1711361111826889?thread_ts=1710492712.079139&cid=C057S9CHMSB

Author: removed

> <div class="alert alert-warning">The full-text search feature is in beta. It is only available in Log Management and works in monitor, dashboard, and notebook queries. The full-text search syntax cannot be used for defining index filters, archive filters, log pipeline filters, or in Live Tail. </div>

Author: ```suggestion
<div class="alert alert-warning">The full-text search feature is only available in Log Management and works in monitor, dashboard, and notebook queries. The full-text search syntax cannot be used for defining index filters, archive filters, log pipeline filters, or in Live Tail. </div>
```

> <div class="alert alert-warning">The full-text search feature is only available in Log Management and works in monitor, dashboard, and notebook queries. The full-text search syntax cannot be used for defining index filters, archive filters, log pipeline filters, or in Live Tail. </div>

Author: > works in monitor, dashboard, and notebook queries.

@siddhingra can you confirm this is correct?

Reviewer 3: yes, this is correct

> <div class="alert alert-warning">You can use full-text search for all logs ingested after MONTH DATE YEAR.</div>

Author: @siddhingra could you let me know the date the write path was turned on?

Reviewer 3: this feels way too prominent, maybe we don't need to call this out at all

Author: removed

Use the syntax `*:search_term` to perform a free text search across all log attributes, including the log message.

> #### Single term example

Reviewer 2: Maybe let's add an example of the difference between `*:kelner` vs `*:*kelner*` ? See these two screenshots

![Screenshot 2024-03-26 at 2 40 07 PM](https://github.com/DataDog/documentation/assets/133862/f55df332-b7fe-45d5-ba9d-de490e0968f6)
vs
![Screenshot 2024-03-26 at 2 40 29 PM](https://github.com/DataDog/documentation/assets/133862/18fe9b53-0e11-45f1-b838-a2f270736087)

1,308 results vs 1,907,477 results

Author: @ckelner Added an example below the single term example, is that what you are looking for?

Author: DM'd this works 🎉
